frontend/src/views/AccountPages/PlansBuyPage/PlansBuyPage.js

```diff
@@ -42,7 +41,24 @@ export default function PlanBuyPage(props) {
 
     const planActive = useSelector(state => state.planActive);
     const { loading, plans, error } = planActive;
+    const userSignin = useSelector(state => state.userSignin);
+    const { userInfo } = userSignin;
     const dispatch = useDispatch();
+    let yearDate = new Date();
+    yearDate = yearDate.setFullYear(yearDate.getFullYear() + 1);
+
+    function checkDate(planDate, yearDate, duration) {
```

Comment:
This might be clearer if what's inside `plans.map` is extracted into it's own component. `checkDate` and `checkDisabled` don't describe what the function does only suggests where they might be used. eg `checkDate` could be  `planExpiresBefore1Year` (although I'm not sure of it's true intention, so this might not be a correct naming).

Reply:
Done